magicctapipe/image/calib.py

```diff
@@ -0,0 +1,134 @@
+                # Scale the charges by the correction factor
+                image *= config["charge_correction"]["factor"]
+            # Apply the image cleaning
+            signal_pixels, image, peak_time = magic_clean[tel_id].clean_image(
+                event_image=image, event_pulse_time=peak_time
+            )
+        else: 
+            raise ValueError("Check the provided magic_clean parameter; calibration was not possible")
+    elif (LST_bool==True) and (obs_id!=None) and (camera_geoms!=None):
+        if type(camera_geoms[tel_id])==CameraGeometry:  
+            increase_nsb = config["increase_nsb"].pop("use")
+            increase_psf = config["increase_psf"]["use"]
+            use_time_delta_cleaning = config["time_delta_cleaning"].pop("use")
+            use_dynamic_cleaning = config["dynamic_cleaning"].pop("use")
+            use_only_main_island = config["use_only_main_island"]
+        
+            if increase_nsb:
+                rng = np.random.default_rng(obs_id)
+                # Add extra noise in pixels
+                image = add_noise_in_pixels(rng, image, **config["increase_nsb"])
+
+            if increase_psf:
+                set_numba_seed(obs_id)
+                # Smear the image
+                image = random_psf_smearer(
+                    image=image,
+                    fraction=config["increase_psf"]["fraction"],
+                    indices=camera_geoms[tel_id].neighbor_matrix_sparse.indices,
+                    indptr=camera_geoms[tel_id].neighbor_matrix_sparse.indptr,
+                )
+
+            # Apply the image cleaning
+            signal_pixels = tailcuts_clean(
+                camera_geoms[tel_id], image, **config["tailcuts_clean"]
+            )
+
+            if use_time_delta_cleaning:
+                signal_pixels = apply_time_delta_cleaning(
+                    geom=camera_geoms[tel_id],
+                    mask=signal_pixels,
+                    arrival_times=peak_time,
+                    **config["time_delta_cleaning"],
+                )
+
+            if use_dynamic_cleaning:
+                signal_pixels = apply_dynamic_cleaning(
+                    image, signal_pixels, **config["dynamic_cleaning"]
+                )
+
+            if use_only_main_island:
+                _, island_labels = number_of_islands(camera_geoms[tel_id], signal_pixels)
+                n_pixels_on_island = np.bincount(island_labels.astype(np.int64))
+
+                # The first index means the pixels not surviving
+                # the cleaning, so should not be considered
+                n_pixels_on_island[0] = 0
+                max_island_label = np.argmax(n_pixels_on_island)
+                signal_pixels[island_labels != max_island_label] = False
+            config["increase_nsb"]["use"]=increase_nsb
+            config["time_delta_cleaning"]["use"]=use_time_delta_cleaning
+            config["dynamic_cleaning"]["use"]=use_dynamic_cleaning
+        else: 
```

Comment:
I think it make the code a bit more clear if you do it with 
if ...bad condition...
  raise ....
if ... another bad conditon
  raise ...
... code if everything is fine...

since raise will immediately stop the execution this way you are avoiding multiple else and multiple indents of the  code. 
but if you prefer it like it is now just close this comment.

Reply:
We can update it, thanks